internal/webhook/daemonsetmutation/webhookhandler.go

```diff
@@ -0,0 +1,95 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package daemonsetmutation contains the webhook that injects annotations into daemon-sets.
+package daemonsetmutation
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/aws/amazon-cloudwatch-agent-operator/internal/config"
+)
+
+// +kubebuilder:webhook:path=/mutate-v1-daemonset,mutating=true,failurePolicy=ignore,groups="apps",resources=daemonsets,verbs=create;update,versions=v1,name=mdaemonset.kb.io,sideEffects=none,admissionReviewVersions=v1
+// +kubebuilder:rbac:groups="",resources=namespaces,verbs=list;watch
```

Comment:
We dont need to list/watch namespaces do we?

Reply:
Yeah well I think this is used to watch namespaces incase the resource is not created yet, I think the pod mutator does something similar - https://github.com/aws/amazon-cloudwatch-agent-operator/blob/main/internal/webhook/podmutation/webhookhandler.go#L66